src/zap/zapinfo.cpp

```diff
@@ -3378,7 +3378,7 @@ CorInfoHelpFunc ZapInfo::getCastingHelper(CORINFO_RESOLVED_TOKEN * pResolvedToke
 CorInfoHelpFunc ZapInfo::getNewArrHelper(CORINFO_CLASS_HANDLE arrayCls)
 {
 	if (IsReadyToRunCompilation())
-		return CORINFO_HELP_NEWARR_1_DIRECT;
+		return CORINFO_HELP_NEWARR_1_R2R_DIRECT;
```

Comment:
Appending of `ELEMENT_TYPE_NATIVE_ARRAY_TEMPLATE_ZAPSIG` in ComputeRuntimeLookupForSharedGenericToken should be skipped for `IsReadyToRunCompilation()` now